### ISSUE TITLE
[Feature] Allow @Destination on other annotation (As a way to define constant deeplinks for all destinations)

### ISSUE THREAD

**Issue by Nek-12** [raamcosta/compose-destinations#346]
I have a case where all deeplinks in an app follow the same pattern known compile-time:

```kotlin
private const val LocalDeeplink = "$Schema://$Path/$FULL_ROUTE_PLACEHOLDER"
private const val RemoteDeeplink = "$Host/$FULL_ROUTE_PLACEHOLDER"
```

This results in same boilerplate code for all destinations that are defined

```kotlin
@Composable
@Destination(
    deepLinks = [
        DeepLink(
            uriPattern = LocalDeeplink,
        ),
        DeepLink(
            uriPattern = RemoteDeeplink,
            action = Intent.ACTION_VIEW,
        )
    ],
)
```

Is there any way to make this universal? I tried defining a constant using `arrayOf()` but kotlin does not allow const val arrays of annotations it seems. Maybe I could create my own annotation with these values hardcoded?

**Comment by raamcosta** [comment]
We could have a feature like this:

```kotlin

@Destination(
    deepLinks = [
        DeepLink(
            uriPattern = LocalDeeplink,
        ),
        DeepLink(
            uriPattern = RemoteDeeplink,
            action = Intent.ACTION_VIEW,
        )
    ],
)
@MyDestination
```

And then use @MyDestination on your screens instead of @Destination.

I'll keep this as future enhancement.

**Comment by raamcosta** [comment]
I am currently on it 🙂 

The challenge is merging info from the multiple annotations since I want to let developers extend a common annotation with different stuff, for example:

```kotlin
@Destination(
    deepLinks = [
        DeepLink(
            uriPattern = LocalDeeplink,
        ),
        DeepLink(
            uriPattern = RemoteDeeplink,
            action = Intent.ACTION_VIEW,
        )
    ],
)
annotation class MyDestination(
    val style: KClass<out DestinationStyle> = DestinationStyle.Default::class
)

@MyDestination(style = DestinationStyle.Dialog::class)
@Composable
fun MyScreen() //...
```

And you could ideally have multiple levels of this, think in previous example MyDestination could use another custom annotation "MySecondDestination". And "MySecondDestination" then is the use that uses `@Destination`.

**Comment by Nek-12** [comment]
Well I'm personally ok with not having parameters and just making multiple annotations like `@BottomSheetDestination`, `@DeeplinkDestination` etc. 
They can be incompatible with each other for the first iteration of the feature.

**Comment by raamcosta** [comment]
I considered that, but I am close to the ideal solution. So might as well finish it 🙂

**Comment by raamcosta** [comment]
So.. Feature is available on 1.7.41-beta, 1.8.41-beta and 1.9.41-beta :) 

Please give it a try and please let me know how it goes 🙏
